pdm: 2.25.5 -> 2.25.9

MisileLab opened the pull request:
https://github.com/pdm-project/pdm/releases/tag/2.25.6
https://github.com/pdm-project/pdm/releases/tag/2.25.7
https://github.com/pdm-project/pdm/releases/tag/2.25.8
https://github.com/pdm-project/pdm/releases/tag/2.25.9

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

applePrincess commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 416496`
Commit: `4cf4701afda2f345bb9ce3117b59123f1bed463b`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>pdm</li>
    <li>pdm.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>pdm</summary>
<pre> - Adding src/pdm/resolver/python.py -> pdm/resolver/python.py
 - Adding src/pdm/resolver/reporters.py -> pdm/resolver/reporters.py
 - Adding src/pdm/resolver/resolvelib.py -> pdm/resolver/resolvelib.py
 - Adding src/pdm/resolver/uv.py -> pdm/resolver/uv.py
 - Adding src/pdm/signals.py -> pdm/signals.py
 - Adding src/pdm/termui.py -> pdm/termui.py
 - Adding src/pdm/utils.py -> pdm/utils.py
 - Adding pdm-2.25.0.dist-info/RECORD -> pdm-2.25.0.dist-info/RECORD
Successfully built pdm-2.25.0-py3-none-any.whl
Finished creating a wheel...
/build/source/dist /build/source
Unpacking to: unpacked/pdm-2.25.0...OK
Repacking wheel as ./pdm-2.25.0-py3-none-any.whl...OK
/build/source
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for pdm-2.25.0-py3-none-any.whl
  - pbs-installer>=2025.6.6 not satisfied by version 2025.3.17</pre>
</details>
</details>

applePrincess commented:
Seems pbs-installer is not up to date. Maybe approve & merge #406382 first?

MisileLab commented:
> Seems pbs-installer is not up to date. Maybe approve & merge #406382 first?

https://github.com/NixOS/nixpkgs/pull/406382

hmm, I can't merge this

natsukium reviewed (commented):
Please move `msgpack` to `optional-dependencies.msgpack`.
https://github.com/pdm-project/pdm/commit/76ebca3c6a1faf749e34d84843be7f26d9d2360e

MisileLab commented:
> > Please move `msgpack` to `optional-dependencies.msgpack`. [pdm-project/pdm@76ebca3](https://github.com/pdm-project/pdm/commit/76ebca3c6a1faf749e34d84843be7f26d9d2360e)
> 
> How can I do that?

nvm, I found

natsukium commented on pkgs/by-name/pd/pdm/package.nix:
```diff
@@ -74,7 +77,8 @@ python.pkgs.buildPythonApplication rec {
     ]
-    ++ httpx.optional-dependencies.socks;
+    ++ httpx.optional-dependencies.socks
+    ++ optional-dependencies.msgpack;
```
`optional-dependencies` should not be added to `dependencies`.
If needed for tests, please add it to `nativeCheckInputs` instead.

MisileLab commented:
cc @natsukium @cpcloud

MisileLab committed:
pdm: 2.25.5 -> 2.25.9

https://github.com/pdm-project/pdm/releases/tag/2.25.6
https://github.com/pdm-project/pdm/releases/tag/2.25.7
https://github.com/pdm-project/pdm/releases/tag/2.25.8
https://github.com/pdm-project/pdm/releases/tag/2.25.9

Signed-off-by: misilelab <misileminecord@gmail.com>

MisileLab commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 416496`

Logs: https://github.com/MisileLab/nixpkgs-review-gha/actions/runs/17329213394


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>pdm</li>
    <li>pdm.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>pdm</li>
    <li>pdm.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>pdm</li>
    <li>pdm.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>pdm</li>
    <li>pdm.dist</li>
  </ul>
</details>

natsukium reviewed (approved):
LGTM
